docs/proposals/20-ha-control-planes.md

```diff
@@ -0,0 +1,2624 @@
+
+**Case #2**
+
+A shoot has a pre-defined non-HA seed. A change has been made to the shoot spec, setting control HA to `multi-zone`. Garden scheduler needs to react to the change in the HA configuration for the shoot control plane.
+
+_Proposed Changes_
+
+* [scheduler manager](https://github.com/gardener/gardener/blob/4dbba56e24b5999fd728cb84b09a4d6d54f64479/pkg/scheduler/controller/shoot/scheduler.go#L57) should remove the predicate `ShootIsUnassigned` and instead define a new predicate which allows a `reconcile.Request` to be created if one of the following is satisified:
+  * Shoot does not have any assigned seed
+  * Shoot spec has changed requesting an uprade from non-HA control plane to a HA control plane. This can be checked by checking the HA configuration in the spec and compare it with the `status` to determine if the upgrade requested is from a non-HA shoot control plane to a HA shoot control plane.
+* [reconciler](https://github.com/gardener/gardener/blob/4dbba56e24b5999fd728cb84b09a4d6d54f64479/pkg/scheduler/controller/shoot/reconciler.go#L52-L99) - function now needs to additionally check if an upgrade request has been made. If it is a valid upgrade request for the shoot control plane then the reconciler should unassign the current seed for this shoot. A new seed appropriate to the HA control plane requirement will be assigned. In this case since there is a change in the seed assignment, `control-plane-migration` flow will kick in and will commence the migration of the control plane from the non-HA seed to a HA seed.
+
+**Unbalanced usage of zones in a multi-zonal seed cluster**
+
+It is possible that `single-zone` shoot control planes are scheduled onto `multi-zonal` seed clusters. This can result in unbalanced utilization of  zones and node groups (in gardener terms this maps to `MachineDeployment`). Optimal zone selection for `single-zone` shoot control plane is currently not possible. Selection of zone and node-group will be best effort.
+
+### Compute Seed Usage
+
+At present seed usage is computed by counting the number of shoot control planes that are hosted in a seed. Every seed has a `status.allocatable.shoots` set to 250 (configurable via [ResourceConfiguration](https://github.com/gardener/gardener/blob/4e2d28f2af7093eb94819652a6f4709b5fbfaf06/pkg/gardenlet/apis/config/v1alpha1/types.go#L449). Operators need to rethink this value for multi-zonal seed clusters.
+
+Which parameters could be considered?
+* Number of available machines of a type as requsted as part of the shoot spec. Sufficient capacity should be available to also allow rolling updates which will also be governed by `maxSurge` configuration at the worker pool level.
+* Node CIDR ranges needs to be adjusted as that will have a bearing on the number of nodes per seed.
```

Comment:
That is not possible without recreating the seed clusters, I believe. Do we have some more data on what needs to happen, either in this document or somewhere else (probably more appropriate since seed cluster changes are out of scope here).